examples/editor/index.js

```diff
@@ -365,3 +369,101 @@ document.getElementById('fullscreen').onclick = () => {
     document.querySelector('.map-info').style.display = 'none';
     map.resize();
 };
+
+const $exportMapButton = document.getElementById('export-map-button');
+const $copyHTMLButton = document.getElementById('copy-html-button');
+const $mapTextarea = document.getElementById('map-textarea');
+
+$exportMapButton.addEventListener('click', () => {
+    const config = getJSONConfig();
+    $mapTextarea.value = generateSnippet(config);
+});
+
+$copyHTMLButton.addEventListener('click', () => {
+    $mapTextarea.select();
+    document.execCommand('copy');
+});
+
+
+/**
+
+
+"a":"sf_stclines",
+"b":"",
+"c":"dmanzanares",
+"d":"carto.com",
+"e":"color: ramp($st_type, prism) \nwidth: 1.5",
+"f":{
+    "lng":-122.44408486861192,
+    "lat":37.773706736149705
+    },
+ "g":11.664310802866805
+}
+
+* @param {*} config 
+ * 
+ */
+function generateSnippet(config) {
+    const dataset = config.a;
+    const apiKey = config.b || 'default_public';
+    const username = config.c;
+    const serverURL = config.d || 'https://{user}.carto.com';
+    const vizSpec = config.e || '';
+    const center = config.f || { lat: 0, lng: 0 };
+    const zoom = config.g || 10;
+    // const basemap = config.h || 'dark-matter';
+
+
+    return `<!DOCTYPE html>
+        <html>
+        <head>
+        <title>Add layer | CARTO</title>
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <meta charset="UTF-8">
+        <!-- Include CARTO VL JS -->
+        <script src="https://cartodb.github.io/carto-vl/dist/carto-vl.js"></script>
+        <!-- Include Mapbox GL JS -->
+        <script src="https://libs.cartocdn.com/mapbox-gl/v0.45.0-carto1/mapbox-gl.js"></script>
+        <!-- Include Mapbox GL CSS -->
+        <link href="https://libs.cartocdn.com/mapbox-gl/v0.45.0-carto1/mapbox-gl.css" rel="stylesheet" />
+        <style>
+           #map {
+                position: absolute;
+                width: 100vw;
+                height: 100vh;
+            }
+        </style>
+        </head>
+        <body>
+        <div id="map"></div>
+
+        <script>
+            const map = new mapboxgl.Map({
+                container: 'map',
+                style: 'https://basemaps.cartocdn.com/gl/voyager-gl-style/style.json',
```

Comment:
We can merge without it, but... it would be awesome if the basemap could be added dynamically here.